### PR TITLE
refactor: Switch support css var

### PR DESCRIPTION
**Pull request by JarvisArt**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Switch support css variables. |
| 🇨🇳 Chinese | Switch 组件支持 css 变量。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 89f075f</samp>

This pull request enhances the Switch component with CSS variables support and style refactoring. It introduces a new `useCSSVar` hook that registers and updates the CSS variables for the component. It also uses the `@ant-design/cssinjs` package to simplify the style calculations and units.

### <samp>🤖 Generated by Copilot at 89f075f</samp>

*  Add the `useCSSVar` hook to enable the use of CSS variables for the Switch component ([link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e6cc84da9314e7733f7658f68135d16dbc896ce22ec302e43189778cc134cf03R11), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-adc35ba68eab1e8f63ac38286783132b6f4bbed966d7b80c2def17272adb209dR1-R4), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e6cc84da9314e7733f7658f68135d16dbc896ce22ec302e43189778cc134cf03L94-R96), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e6cc84da9314e7733f7658f68135d16dbc896ce22ec302e43189778cc134cf03L117-R119))
*  Import the `unit` function and the `GetDefaultToken` type to convert numeric values to CSS units and annotate the `prepareComponentToken` function ([link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L2-R5))
*  Replace the numeric values and template literals with the `calc` function calls to simplify the arithmetic expressions and ensure valid CSS units in the Switch component styles ([link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L93-R100), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L101-R106), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L107-R113), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L128-R129), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L142-R149), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L159-R157), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L166-R164), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L183-R181), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L216-R214), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L224-R222), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L246-R257), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L270-R283), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L294-R299), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L306-R307), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L313-R314), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L334-R334))
*  Extract the `prepareComponentToken` function to calculate the token values for the Switch component from the `useSwitchStyle` hook in `components/switch/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571R376-R401), [link](https://github.com/ant-design/ant-design/pull/45867/files?diff=unified&w=0#diff-e2a7c264e7542fee20fd50cc83119edf369c978f283c2e0d80675fffccfb4571L404-R430))

### CODE REVIEW

File: components/switch/style/index.ts [hunk 1]
```diff
@@ -401,29 +427,5 @@ export default genComponentStyleHook(
       genSwitchSmallStyle(switchToken),
```
Comment (MadCcc) [inline components/switch/style/index.ts:427]: switchToken 里有个 switchLoadingIconSize 计算过

Comment (JarvisArt) [reply components/switch/style/index.ts:426]: done~

**Comment by MadCcc** [comment]
点击时的动画没了

**Comment by MadCcc** [comment]
image
有三处 NaN

**Comment by JarvisArt** [comment]
> 点击时的动画没了

> image 有三处 NaN

会不会是之前的预览链接，点现在链接康康？

Uploading 屏幕录制2023-11-14 16.38.53.mov…

**Comment by JarvisArt** [comment]
> > 点击时的动画没了
> 
> > image 有三处 NaN
> 
> 会不会是之前的预览链接，点现在链接康康？


https://github.com/ant-design/ant-design/assets/38420763/ab609eb0-ff1c-4902-baa9-bd8cf7d63e26

**Comment by MadCcc** [comment]
@JarvisArt check-cssinjs 的 switch 去掉跑一下 lint 看看？

**Comment by MadCcc** [comment]
点击的时候有一个白球拉长的动画

**Comment by JarvisArt** [comment]
> 点击的时候有一个白球拉长的动画

确实

File: components/switch/style/index.ts [hunk 2]
```diff
@@ -213,15 +210,15 @@ const genSwitchHandleStyle: GenerateStyle<SwitchToken, CSSObject> = (token) => {
...
           content: '""',
         },
       },
 
       [`&${componentCls}-checked ${switchHandleCls}`]: {
-        insetInlineStart: `calc(100% - ${handleSize + trackPadding}px)`,
+        insetInlineStart: `calc(100% - ${unit(calc(handleSize).add(trackPadding).equal())})`,
       },
 
       [`&:not(${componentCls}-disabled):active`]: motion
```
Comment (JarvisArt) [inline components/switch/style/index.ts:224]: 待处理 @MadCcc

**Review by MadCcc — APPROVED** [review]

**Pull request merged**